dart/lib/src/hub.dart

```diff
@@ -0,0 +1,297 @@
+import 'dart:async';
+import 'dart:collection';
+
+import 'client.dart';
+import 'noop_client.dart';
+import 'protocol.dart';
+import 'scope.dart';
+import 'sentry_options.dart';
+
+typedef ScopeCallback = void Function(Scope);
+
+/// SDK API contract which combines a client and scope management
+class Hub {
+  static SentryClient _getClient({SentryOptions fromOptions}) {
+    return SentryClient(
+      dsn: fromOptions.dsn,
+      environmentAttributes: fromOptions.environmentAttributes,
+      compressPayload: fromOptions.compressPayload,
+      httpClient: fromOptions.httpClient,
+      clock: fromOptions.clock,
+      uuidGenerator: fromOptions.uuidGenerator,
+    );
+  }
+
+  final ListQueue<_StackItem> _stack;
```

Comment:
we use a `Deque` in Java, we'd need to learn the differences if not a Deque available in Dart

Reply:
Dart `ListQueue` seems to offer the same features

Reply:
that means adding a dependency, could we reconsider adding this code right now until we are certain we'll add push/pop scope?

Reply:
its from `dart:collection` (standard lib ) , not `pkg:collection`